### PR TITLE
feat: add installation script

### PR DESCRIPTION
**Pull request by chaitanyabsprip**
## Purpose
There should be an easy way to automate installation of gitmux.
#104 

## Approach
Currently I've in-lined all dependencies from my scripts and found an alternative method of extracting latest version from Github api instead of using jq.

### CODE REVIEW

**Comment by Chaitanyabsprip** [comment]
I'm not sure how the version argument can be respected when downloading from a package manager. I haven't added choice of version if the package is being installed from a package manager. I would suggest dropping that part of the script and relying only on curl and github api.

**Comment by Chaitanyabsprip** [comment]
There needs to be a discussion about what the default installation directory will be. Currently I've kept it as `$HOME/.local/bin`

File: install [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    Arch\ Linux) paru -S --noconfirm gitmux ;;
+    Darwin)
+        brew tap arl/arl
+        brew install gitmux
+        ;;
+    *) error "Unsupported platform" && return 1 ;;
+    esac
+}
+
+go_install() { :; }
```
Comment (arl) [inline install:34]: Not needed

File: install [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        DISTRO=$(lsb_release -si)
+        VERSION=$(lsb_release -sr)
+    else
+        DISTRO=$(uname -s)
+        VERSION=$(uname -r)
+    fi
+    echo "$DISTRO:$VERSION"
+}
+
+pkg_install() {
```
Comment (arl) [inline install:23]: Not needed

File: install [hunk 3]
```diff
@@ -0,0 +1,71 @@
+#!/bin/sh
+
+success() { printf " \033[1;32m ✔ %s\033[0m\n" "$*"; }
```
Comment (arl) [inline install:3]: Pretty printing/formatting not needed for this script.
We should give a clear error message and exit 1 or 0 based on the final outcome

File: install [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+inprogress() { printf " \033[1;34m ... %s\033[0m\n" "$*"; }
+warning() { printf " \033[1;33m ⚠️%s\033[0m\n" "$*"; }
+error() { printf " \033[1;31m ❌%s\033[0m\n" "$*"; }
+distro() {
```
Comment (arl) [inline install:7]: out of scope

File: install [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        brew tap arl/arl
+        brew install gitmux
+        ;;
+    *) error "Unsupported platform" && return 1 ;;
+    esac
+}
+
+go_install() { :; }
+
+gh_install() {
```
Comment (arl) [inline install:36]: this is the part we need

**Review by arl — CHANGES_REQUESTED** [review]
> I'm not sure how the version argument can be respected when downloading from a package manager. I haven't added choice of version if the package is being installed from a package manager. I would suggest dropping that part of the script and relying only on curl and github api.

I think we should refine and limit the scope of the script. I expect the script to be used by people that wants to automate installation for their dot files, not to install gitmux for any package manager/distro out there.

In that case, the user already knows the os and architecture they need, these should be passed as arguments to the script.

As for the version number, for a first version of the script I'd say we can also pass it as parameter, and refine later.



> There needs to be a discussion about what the default installation directory will be. Currently I've kept it as `$HOME/.local/bin`

I think this should also be passed as parameter to the script.

To sum it up: we should keep it very simple:
 - check that all parameters are passed
 - check that dependency exist (curl or wget, jq)

This should be already helpful for most people.
And it'd be a good base if more features/automation is needed later.

**Comment by Chaitanyabsprip** [comment]
I've made the requested changes.

**Review by arl — APPROVED** [review]
Sorry for the long time it took me.
That looks good as a first step towards providing a one-liner install like many projects. 

In a next PR, I'd like to add a GitHub action that runs your script for 2 prominent os/arch pairs, after release creation, so we can be confident to propose this directly in the readme.

Thanks a lot @Chaitanyabsprip

**Review by arl — APPROVED** [review]
LGTM!
